### PR TITLE
TVIST1-606: Ensured unlinking of temp files

### PR DESCRIPTION
**Pull request by jekuaitk**
https://jira.itkdev.dk/browse/TVIST1-606

* Ensures temp files are unlinked

### CODE REVIEW

File: src/Service/DocumentUploader.php [hunk 1]
```diff
@@ -73,11 +73,15 @@ public function createDocumentFromUploadedFile(UploadedFile $file, string $docum
...
 
+        $tempPath = $file->getRealPath();
+
         $document = $this->createDocumentFromPath($file->getRealPath(), $documentName, $documentType);
 
         // Explicitly indicate that this is created manually, i.e. display original file name on document index page.
         $document->setIsCreatedManually(true);
 
+        unlink($tempPath);
+
```
Comment (rimi-itk) [inline src/Service/DocumentUploader.php:84]: Consider using `try … finally` here (cf. https://www.php.net/manual/en/language.exceptions.php#language.exceptions.finally) to make sure that the file is actually deleted even if something goes wrong when creating the document, i.e.

```php
try {
  $tempPath = …
  …
} finally {
  unlink($tempPath);
}
```

File: src/Service/OS2Forms/SubmissionNormalizers/AbstractSubmissionNormalizer.php [hunk 2]
```diff
@@ -110,6 +110,8 @@ private function handleDocuments(string $sender, array $documentIds): array
                 // Create document
                 $document = $this->documentUploader->createDocumentFromPath($newFilePath, basename($documentUrl), 'OS2Forms');
 
+                unlink($newFilePath);
+
```
Comment (rimi-itk) [inline src/Service/OS2Forms/SubmissionNormalizers/AbstractSubmissionNormalizer.php:114]: Use `try … finally` here.

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. Minor change to make the code more robust suggested.

File: src/Service/OS2Forms/SubmissionNormalizers/AbstractSubmissionNormalizer.php (hunk 2 above)
Comment (jekuaitk) [reply src/Service/OS2Forms/SubmissionNormalizers/AbstractSubmissionNormalizer.php]: Updated in b9c8983570afa1efd12f68ce397d899315fb2fe1.

File: src/Service/DocumentUploader.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/DocumentUploader.php]: Good idea. Updated in b9c8983570afa1efd12f68ce397d899315fb2fe1.

File: src/Service/DocumentUploader.php [hunk 3]
```diff
@@ -70,13 +70,22 @@ public function createDocumentFromPath(string $filePath, string $documentName, s
...
-        // Explicitly indicate that this is created manually, i.e. display original file name on document index page.
-        $document->setIsCreatedManually(true);
+            $document = $this->createDocumentFromPath($file->getRealPath(), $documentName, $documentType);
+
+            // Explicitly indicate that this is created manually, i.e. display original file name on document index page.
+            $document->setIsCreatedManually(true);
+        } finally {
+            $myfile = fopen('testfile.txt', 'w');
+            fwrite($myfile, $tempPath);
+            fclose($myfile);
```
Comment (rimi-itk) [inline src/Service/DocumentUploader.php:86]: You may want to remove this …

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Service/DocumentUploader.php (hunk 3 above)
Comment (jekuaitk) [reply src/Service/DocumentUploader.php]: Yep.. whoopsie f71f3751b687960645c385b601ad00f70c259ca1

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**